src/server/controllers/user.js

```diff
@@ -8,27 +8,42 @@ const jwtSecret = 'mysecret';
 const register = async (req, res) => {
     const { username, password } = req.body;
 
-    const createdUser = null;
-
-    res.json({ data: createdUser });
-};
+        const hashedPassword = await bcrypt.hash(password, 10)
+        const createdUser = await prisma.user.create({
+          data: {
+            username: username,
+            password: hashedPassword
+          },
+        });
+      
+        res.json({ data: createdUser });
+      };
+
+//     const createdUser = null;
+
+//     res.json({ data: createdUser });
+// };
 
 const login = async (req, res) => {
     const { username, password } = req.body;
 
-    const foundUser = null;
+    const foundUser = await prisma.user.findFirst({
+      where: {
+        username
+      }
+    });
 
     if (!foundUser) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const passwordsMatch = false;
+    const passwordsMatch = await bcrypt.compare(password, foundUser.password);
 
     if (!passwordsMatch) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
 
-    const token = null;
+    const token = jwt.sign({username}, jwtSecret);
```

Comment:
I'm not sure how you were able to create two const variables with the same name...